huaweicloud/services/mpc/resource_huaweicloud_mpc_transcoding_template.go

```diff
@@ -0,0 +1,503 @@
+	}
+
+	templateList := *resp.TemplateArray
+	template := templateList[0].Template
+	if template == nil {
+		log.Printf("unable to retrieve MPC transcoding template: %s", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", config.GetRegion(d)),
+		d.Set("name", template.TemplateName),
+		d.Set("common", flattenCommon(template.Common)),
+		d.Set("audio", flattenAudio(template.Audio)),
+		d.Set("video", flattenVideo(template.Video)),
+	)
+
+	if err = mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("error setting MPC transcoding template fields: %s", err)
+	}
+
+	return nil
+}
+
+func resourceMPCTranscodingTemplateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcMpcV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating MPC client : %s", err)
+	}
+
+	id, err := strconv.ParseInt(d.Id(), 10, 32)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updateOpts := mpc.ModifyTransTemplateReq{
+		TemplateId:   id,
+		TemplateName: d.Get("name").(string),
+		Video:        buildVideoOpts(d.Get("video").([]interface{})),
+		Audio:        buildAudioOpts(d.Get("audio").([]interface{})),
+		Common:       buildCommonOpts(d.Get("common").([]interface{})),
+	}
+
+	updateReq := mpc.UpdateTransTemplateRequest{
+		Body: &updateOpts,
+	}
+
+	_, err = client.UpdateTransTemplate(&updateReq)
+	if err != nil {
+		return diag.Errorf("error updating MPC transcoding template: %s", err)
+	}
+
+	return resourceMPCTranscodingTemplateRead(ctx, d, meta)
+}
+
+func resourceMPCTranscodingTemplateDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcMpcV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating MPC client : %s", err)
+	}
+
+	id, err := strconv.ParseInt(d.Id(), 10, 32)
+	if err != nil {
+		diag.FromErr(err)
+	}
+
+	_, err = client.DeleteTemplate(&mpc.DeleteTemplateRequest{TemplateId: id})
+	if err != nil {
+		return fmtp.DiagErrorf("error deleting MPC transcoding template: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func flattenCommon(common *mpc.Common) []map[string]interface{} {
+	if common == nil {
+		return nil
+	}
+	commonResult := make(map[string]interface{})
+	commonResult["pvc"] = common.Pvc
+	commonResult["hls_interval"] = common.HlsInterval
+	commonResult["dash_interval"] = common.DashInterval
+	commonResult["pack_type"] = common.PackType
+
+	return []map[string]interface{}{commonResult}
+}
+
+func flattenAudio(audio *mpc.Audio) []map[string]interface{} {
+	if audio == nil {
+		return nil
+	}
+	audioResult := make(map[string]interface{})
+	audioResult["codec"] = audio.Codec
+	audioResult["sample_rate"] = audio.SampleRate
+	audioResult["channels"] = audio.Channels
+	audioResult["bitrate"] = audio.Bitrate
```

Comment:
```
audioResult := map[string]interface{}{
	"codec":       audio.Codec,
	"sample_rate": audio.SampleRate,
	"channels":    audio.Channels,
	"bitrate":     audio.Bitrate,
}
```
It's easier to read.
Code lines are clearly identified even when panic is triggered.

Reply:
fixed